src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -322,6 +323,10 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
                 {
                     return CommonErrors.CustomError(IRContext.NotInSource(info.RetType), customFunctionErrorException.Message);
                 }
+                catch (ServiceUnavailableException serviceUnavailableException)
+                {
+                    return FormulaValue.NewError(serviceUnavailableException.ExpressionError);
+                }

```

Comment:
can we avoid new exception type and reuse CustomFunctionErrorException with ErrorValue?